lib/req_athena.ex

```diff
@@ -304,6 +311,29 @@ defmodule ReqAthena do
     end
   end
 
+  @credential_keys ~w(access_key_id secret_access_key region token)a
+
+  defp maybe_put_aws_credentials(request) do
+    credentials = get_credentials(request.options)
+    options = Map.take(credentials, @credential_keys)
+
+    %{request | options: Map.merge(request.options, options)}
+  end
+
+  defp get_credentials(options) do
+    credentials_from_opts = Map.take(options, @credential_keys)
+
+    if Code.ensure_loaded?(:aws_credentials) and Process.whereis(:aws_credentials_sup) do
```

Comment:
I was thinking we would do this:

```suggestion
    if Code.ensure_loaded?(:aws_credentials) do
      Application.ensure_all_started(:aws_credentials)
```

This way, if the app is not yet running, we will start it and we can always add `{:aws_credentials, "~> 0.4", runtime: false}` to Mix.install. WDYT?